code/modules/mob/living/say.dm

```diff
@@ -93,6 +93,12 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	if(!message || message == "")
 		return
 
+	if(GLOB.in_character_filter.len && !forced)
+		var/regex/filter = regex("\\b([jointext(GLOB.in_character_filter, "|")])\\b", "i")

```

Comment:
you need to compile this once at world init please

Reply:
Done.